alacritty/src/window_context.rs

```diff
@@ -377,37 +370,55 @@ impl WindowContext {
         self.update_config(config);
     }
 
+    /// Draw the window.
+    pub fn draw(&mut self, scheduler: &mut Scheduler) {
+        self.display.window.requested_redraw = false;
+
+        if self.occluded {
+            return;
+        }
```

Comment:
Shouldn't the compositor handle this by not sending frames?

Reply:
It's more complicated across multiple platforms. Like surely it works like that on Wayland, but not on macOS iirc.